allauth_2fa/forms.py

```diff
@@ -31,59 +27,24 @@ class TOTPDeviceForm(forms.Form):
         label=_("Token"),
     )
 
-    def __init__(self, key, user, metadata=None, **kwargs):
+    def __init__(self, user, metadata=None, **kwargs):
         super(TOTPDeviceForm, self).__init__(**kwargs)
         self.fields['token'].widget.attrs.update({'autofocus': 'autofocus'})
-        self.key = key
-        self.tolerance = 1
-        self.t0 = 0
-        self.step = 30
-        self.drift = 0
-        self.digits = 6
         self.user = user
         self.metadata = metadata or {}
 
     def clean_token(self):
-        try:
-            token = int(self.cleaned_data.get('token'))
-        except ValueError:
-            # valid will never equal true in this case.
-            token = None
-        valid = False
-        t0s = [self.t0]
-        key = unhexlify(self.key.encode())
-        if 'valid_t0' in self.metadata:
-            t0s.append(int(time()) - self.metadata['valid_t0'])
-        for t0 in t0s:
-            for offset in range(-self.tolerance, self.tolerance):
-                expected_token = totp(
-                    key,
-                    self.step,
-                    t0,
-                    self.digits,
-                    self.drift + offset
-                )
-
-                if expected_token == token:
-                    self.drift = offset
-                    self.metadata['valid_t0'] = int(time()) - t0
-                    valid = True
+        token = self.cleaned_data.get('token')
```

Comment:
I think I had added the `try-except` here to catch an error that was being thrown back to users. I'll check this before merging.